From 566982c0aa47cfa233276c741685384849ec364e Mon Sep 17 00:00:00 2001
From: "Kevin F. Haggerty" <haggertk@lineageos.org>
Date: Sat, 25 Aug 2018 20:23:19 -0600
Subject: [PATCH 10/11] soong_config: Allow extension of valid gralloc 1.0
 buffer usage bits

* fw/native change I7f4174581e24e361577640b9263514a168ed482d
  implemented validation of the buffer description info prior to
  creating the descriptor. Some of our devices need to whitelist
  additional usage bits to support camera functionality.
* The TARGET_ADDITIONAL_GRALLOC_10_USAGE_BITS variable can contain
  a singular roll-up value (e.g., 0x02000000U), a left-shifted bit
  (e.g., (1 << 25)), a bitwise OR of these things, or essentially
  anything that the compiler can choke down.

Change-Id: I2127d33b03426b5e0f981aae837e07d82163fa17
---
 build/soong/android/variable.go | 4 ++++
 build/soong/soong_config.mk     | 6 ++++++
 2 files changed, 10 insertions(+)

diff --git a/build/soong/android/variable.go b/build/soong/android/variable.go
index a08a61d2..cbc2a0d2 100644
--- a/build/soong/android/variable.go
+++ b/build/soong/android/variable.go
@@ -1,5 +1,8 @@
 package android
 type Product_variables struct {
+	Additional_gralloc_10_usage_bits struct {
+		Cppflags []string
+	}
 	Has_legacy_camera_hal1 struct {
 		Cflags []string
 	}
@@ -28,6 +31,7 @@ type Product_variables struct {
 }
 
 type ProductVariables struct {
+	Additional_gralloc_10_usage_bits  *string `json:",omitempty"`
 	Has_legacy_camera_hal1  *bool `json:",omitempty"`
 	Java_Source_Overlays *string `json:",omitempty"`
 	Needs_legacy_camera_hal1_dyn_native_handle  *bool `json:",omitempty"`
diff --git a/build/soong/soong_config.mk b/build/soong/soong_config.mk
index 61861442..7ae37870 100644
--- a/build/soong/soong_config.mk
+++ b/build/soong/soong_config.mk
@@ -1,8 +1,14 @@
 _contents := $(_contents)    "Lineage":{$(newline)
 
 # See build/core/soong_config.mk for the add_json_* functions you can use here.
+ifeq ($(TARGET_ADDITIONAL_GRALLOC_10_USAGE_BITS),)
+  $(call add_json_str,  Additional_gralloc_10_usage_bits, 0)
+else
+  $(call add_json_str,  Additional_gralloc_10_usage_bits, $(TARGET_ADDITIONAL_GRALLOC_10_USAGE_BITS))
+endif
 $(call add_json_bool, Has_legacy_camera_hal1,                $(filter true,$(TARGET_HAS_LEGACY_CAMERA_HAL1)))
 $(call add_json_str,  Java_Source_Overlays,                  $(JAVA_SOURCE_OVERLAYS))
+$(call add_json_bool, Has_legacy_camera_hal1,                     $(filter true,$(TARGET_HAS_LEGACY_CAMERA_HAL1)))
 $(call add_json_bool, Needs_legacy_camera_hal1_dyn_native_handle, $(filter true,$(TARGET_NEEDS_LEGACY_CAMERA_HAL1_DYN_NATIVE_HANDLE)))
 $(call add_json_bool, Needs_text_relocations,                $(filter true,$(TARGET_NEEDS_PLATFORM_TEXT_RELOCATIONS)))
 $(call add_json_str,  Specific_camera_parameter_library,     $(TARGET_SPECIFIC_CAMERA_PARAMETER_LIBRARY))
-- 
2.17.1

